### PR TITLE
Travis intregation

### PR DESCRIPTION
**Pull request by AnBucquet**
Related to #127 

**Update:**

**Done**
- add the repository to Travis
- add .travis.yml
- work on Linux
- python 3.5/3.6/3.7 
- OSX (10.12/10.13/10.14)

**TODO**
- Integration test
- Transfer Github actions to Travis

**N.B.**
As the new version of **Pillow** (7.0.0) is not compatible with our version of **Torchvision** (because **_Pillow_version_** has been removed), I.ve pinned the version of **Pillow** to 6.1 to fix this bug.

### CODE REVIEW

**Comment by AnBucquet** [comment]
**Update:**

**Done**
- add the repository to Travis
- add .travis.yml
- work on Linux with python 3.6/3.7
- Fix Pillow bug (_Pillow_version_ has been removed in Pillow 7.0.0)

**WIP**

- Transfer Github actions to Travis
- Fix problems with OSX

**TODO**
- Integration test
- Coverage

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# testing file for Travis
```
Comment (jcohenadad) [inline .travis.yml]: @AnBucquet could you update the PR body (it says "WIP: add .travis.yml-- which seems done)

File: requirements.txt [hunk 2]
```diff
@@ -10,5 +10,6 @@ torch==1.2.0
 torchvision==0.2.1
 tqdm>=4.30
 future
+pillow==6.1.0
```
Comment (jcohenadad) [inline requirements.txt]: is that related to this PR?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    #  language: shell
+
+before_install:
+  - python3 --version
+  - pip3 install -U pip
+  - pip3 install -U pytest
+  - pip3 install codecov
+
+install:
+  - pip3 install -e .
```
Comment (jcohenadad) [inline .travis.yml]: the doc asks to use pip (not pip3)-- i would replicate what is in the doc (the purpose of this testing is to make sure that what a user is instructed to do will work in the long term)

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    - os: linux  # list of Linux env: https://docs.travis-ci.com/user/reference/overview/
+      name: "Ubuntu 18.04 (Bionic Beaver)"
+      dist: bionic
+    - os: linux
+      name: "Ubuntu 16.04 (Xenial)"
+      dist: xenial
+    - os: linux
+      name: "Ubuntu 14.04 (Trusty)"
+      dist: trusty
+    #- os: osx
```
Comment (jcohenadad) [inline .travis.yml]: any reason to remove osx?

Comment (AnBucquet) [reply .travis.yml]: Seems that most **OSX** distributions on Travis don't have python 3.X by default (waiting for Catalina...). For some unknown reasons, _xcode9.4_ works with python3 but tests return a timeout error while **Linux** tests pass.

Trying to fix this. Any help is welcomed

File: .travis.yml (hunk 1 above)
Comment (AnBucquet) [reply .travis.yml]: Done

File: requirements.txt (hunk 2 above)
Comment (AnBucquet) [reply requirements.txt]: It fixed a bug with a new version of **Pillow** (which leads tests to fail). I can move it to another PR if you prefer.

File: .travis.yml (hunk 4 above)
Comment (jcohenadad) [reply .travis.yml]: > tests return a timeout error
i would start looking at where is the lag (which test?) and then try to reduce it. Travis has a maximum running time for processes.

File: requirements.txt (hunk 2 above)
Comment (jcohenadad) [reply requirements.txt]: > I can move it to another PR if you prefer.

it's fine, but just mention it in the body of the PR so reviewers understand this is not related to the main purpose of this PR

File: .travis.yml (hunk 3 above)
Comment (AnBucquet) [reply .travis.yml]: **pip** works on Linux but not on **OSX/ Windows** as it points to **Python 2.X**.
Any advice?

Comment (jcohenadad) [reply .travis.yml]: did you look into these posts?
- https://github.com/travis-ci/travis-ci/issues/9744#issuecomment-426894873
- https://travis-ci.community/t/osx-default-environment-with-python-doesnt-work/6116/2

another easy solution would be to install python locally via distribs like minconda
